### PR TITLE
Add Schema Validator

### PR DESCRIPTION
**Pull request by fpenim**
https://www.ebi.ac.uk/panda/jira/browse/SUBS-1289
This PR adds the integration for the JSON Schema Validation. 

This is what the schema validator does:
1. Receives validation request;
2. Fetches the json schema for the submittable being validated;
3. Calls the [json-schema-validator](https://github.com/EMBL-EBI-SUBS/json-schema-validator), sending the _schema_ and _submittable_;
4. Gets the validation errors back from the [json-schema-validator](https://github.com/EMBL-EBI-SUBS/json-schema-validator);
5. Sends the results to the appropriate routing key to be received by the aggregator.

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/validator/schema/JsonSchemaValidationHandler.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        List<SingleValidationResult> singleValidationResultList;
+        if(jsonSchemaValidationErrors.isEmpty()) {
+            singleValidationResultList = Arrays.asList(generatePassingSingleValidationResult(envelope.getEntityToValidate().getId(), ValidationAuthor.JsonSchema));
+        } else {
+            singleValidationResultList = covertToSingleValidationResultList(jsonSchemaValidationErrors, envelope.getEntityToValidate().getId());
+        }
+        return singleValidationResultList;
+    }
+
+    private List<SingleValidationResult> covertToSingleValidationResultList(List<JsonSchemaValidationError> errorList, String entityUuid) {
```
[the remark this answers is not in the document]
Comment (fpenim) [reply src/main/java/uk/ac/ebi/subs/validator/schema/JsonSchemaValidationHandler.java]: 👍

File: src/main/java/uk/ac/ebi/subs/validator/schema/custom/CustomHttpMessageConverter.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+
+import java.util.Arrays;
+
+/**
+ * The GitHub raw endpoint serves the Content-Type as text/plain, for that reason, knowing that the endpoints
+ * we will be calling are in fact returning a json response body we need a custom HttpMessageConverter that is in fact
+ * the same as the MappingJackson2HttpMessageConverter but accepts Content-Type as text/plain.
+ */
+public class CustomHttpMessageConverter extends MappingJackson2HttpMessageConverter {
```
[the remark this answers is not in the document]
Comment (fpenim) [reply src/main/java/uk/ac/ebi/subs/validator/schema/custom/CustomHttpMessageConverter.java]: 1. Yes, I can rename it to something more meaningful.
2. It's already being added as an extra converter, I'm not overriding any of the converters or media types. See the `SchemaService` [constructor](https://github.com/EMBL-EBI-SUBS/validation-service/blob/SUBS-1289-schema-validator/src/main/java/uk/ac/ebi/subs/validator/schema/SchemaService.java#L18).

File: src/main/java/uk/ac/ebi/subs/validator/schema/SchemaService.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        messageConverters.add(new CustomHttpMessageConverter());
+        this.restTemplate.setMessageConverters(messageConverters);
+    }
+
+    public JsonNode getSchemaFor(String submittableType, String schemaUrl) throws SchemaNotFoundException {
+        JsonNode schema;
+        try {
+            schema = restTemplate.getForObject(schemaUrl, ObjectNode.class);
+        } catch (RestClientException e) {
+            throw new SchemaNotFoundException(submittableType + " schema - " + e.getMessage(), e);
```
[the remark this answers is not in the document]
Comment (fpenim) [reply src/main/java/uk/ac/ebi/subs/validator/schema/SchemaService.java]: Yes, that is the point of this custom exception. The name says exactly what is happening. We are failing to get the schema and the exception has the detailed information of the schema we are trying to get and the cause exception that prevented us from finding it.

File: src/main/java/uk/ac/ebi/subs/validator/schema/model/JsonSchemaValidationError.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            if (i == 0) {
+                errorString = errors.get(i);
+            } else {
+                errorString += ", " + errors.get(i);
+            }
+
+            if(i == errors.size()-1) {
+                errorString += ".";
+            }
+        }
```
[the remark this answers is not in the document]
Comment (fpenim) [reply src/main/java/uk/ac/ebi/subs/validator/schema/model/JsonSchemaValidationError.java]: That is cleaner, cool 👍

File: src/main/java/uk/ac/ebi/subs/validator/schema/SchemaService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/validator/schema/SchemaService.java]: Yes, but the name "not found" would tell me that the required schema is not exists, but that not always true when we wrap a `RestClientException`.

File: src/main/java/uk/ac/ebi/subs/validator/schema/custom/CustomHttpMessageConverter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/validator/schema/custom/CustomHttpMessageConverter.java]: 1. `PlanTextHttpMessageConverter` ?
2. OK. I missed that constructor.

**Pull request merged**